process-dataset/steps/upload-dataset-and-manifest.js

```diff
@@ -0,0 +1,44 @@
+const fsPromises = require('fs').promises;
+const dataPrep = require("../data-validation/data-prep");
+const schemas = require("../data-validation/schema");
+
+const uploadDatasetAndManifest = async (firebaseHandler, datasetJson, readFolder, featureDefsFileName) => {
+    console.log("uploading dataset description and manifest...");
+    const readFeatureData = async () => {
+        const data = await fsPromises.readFile(`${readFolder}/${featureDefsFileName}`)
+        return JSON.parse(data)
+    }
+
+    const featureData = await readFeatureData();
+    const dataset = dataPrep.initialize(datasetJson, schemas.datasetSchema)
+
+    const manifest = dataPrep.initialize(datasetJson, schemas.manifestSchema)
+    // will be updated when the data is uploaded
+    manifest.cellLineData = "";
+    manifest.albumPath = "";
+    manifest.featuresData = "";
+
+    manifest.featuresDisplayOrder = featureData.map(ele => ele.key)
+    const datasetCheck = dataPrep.validate(dataset, schemas.dataset)
+    const manifestCheck = dataPrep.validate(manifest, schemas.manifest)
+    if (datasetCheck.valid) {
+        // upload dataset
+        await firebaseHandler.uploadDatasetDoc(dataset)
+    } else {
+        console.log(datasetCheck.error);
+        process.exit(1);
+    }
+    if (manifestCheck.valid) {
+        // upload manifest
+        await firebaseHandler.uploadManifest(manifest)
```

Comment:
Does it leave things in a bad state on the remote if one of these intermediate steps fails?

Reply:
Not these ones, no writing to anything has happened yet, so I tried to have it error and exit if anything was missing in the init files.